HealthyPlanners/Urban-Sherpa
# Universal form

**Tasks**
-Created reusable Form component that can manage error handling and onChange and submit events
-Created reusable input component to be used with new Form component
-Created reusable button component to be used with new Form component
-Created reusable select component to be used with new Form component
-Created "TestLogin" component solely used for testing the new reusable components
-Deleted some unused / empty files
-Imported @hapi/joi to be used as default for reusable component (though not required)

**To Test**
-Import and render "TestLogin" from client/components/common/experimental as the first element in App.js
-You should now see a form with three text inputs, a select input, and a submit button. 
-Try submitting without entering data and errors should render under each input. 
-Navigate to the "TestLogin" component.
-Uncomment line 44 where the "checkErrorsOnSubmitOnly" is set to false. 
-Now enter data in the TestLogin form in the browser. 
-Errors should render as you input data. 
-The Submit button should be disabled until all data is error free
-Click Submit
-The console should log "submitted" along with your data object

**The Files**
**Form.jsx**
-This file looks like I deleted and added many lines of code. There was an unused Form.jsx which I initially renamed to ClassForm. Then this Form.jsx component was created from scratch. In short the deletions mean nothing. This is meant to be highly configurgable. Use the propTypes and comments as your guide.

**FormButton.jsx**
-This is a button component that can optionally be rendered as a child to the Form component. Use the propTypes as your guide.

**FormInput.jsx**
-Meant as highly configurable Input component to be rendered as child to Form component as needed. Renders errors passed by Form component and uses state from Form component as its value.

**FormSelect.jsx**
-Meant as highly configurable Input component to be rendered as child to Form component. Follows same pattern as FormInput

**Input.jsx**
-Unused component -- Part of old Form and deleted

**Suspend.jsx**
-No substantive changes here. This just got caught up in a PR in which it didn't belong. I renamed ONE variable and fixed a linting error by explicitly returning undefined

**TestLogin.jsx**
-Created simply to test the reusable form component. Should be deleted when all are satisfied that this works

**Placeholder.jsx**
-Empty File - Deleted

**LongLatTest**
-Handled linting errors. Removed unused imports and it looks like prettier had its way with single parameter arrow functions

**package-Lock and package.json**
-Imported @hapi-joi to facilitate default useage of common Form component
